### ISSUE TITLE
Panic in Metricbeat Redis module

### ISSUE THREAD

**Issue by jwarren116** [elastic/beats#13944]
I previously reported a bug on the forums @ https://discuss.elastic.co/t/panic-in-redis-module/197253. The bug was reported as fixed in 6.8.3 by https://github.com/elastic/beats/pull/13426, but I am still seeing the same panic. Maybe this is an additional issue or edge case?

- Version: 6.8.3
- Operating System: Ubuntu 16.04
- Discuss Forum URL: https://discuss.elastic.co/t/panic-in-redis-module/197253
- Steps to Reproduce: See below.

Error from logs:
```
2019-10-04T06:09:25.988Z	ERROR	runtime/hashmap_fast.go:696	recovered from panic while fetching 'redis/key' for host 'xxxxxxx:6379'. Recovering, but please report this.
	{"panic": "assignment to entry in nil map", "stack": "github.com/elastic/beats/libbeat/logp.Recover
    /go/src/github.com/elastic/beats/libbeat/logp/global.go:105
runtime.call32
    /usr/local/go/src/runtime/asm_amd64.s:573
runtime.gopanic
    /usr/local/go/src/runtime/panic.go:502
runtime.mapassign_faststr
    /usr/local/go/src/runtime/hashmap_fast.go:696
github.com/elastic/beats/metricbeat/module/redis/key.eventMapping
    /go/src/github.com/elastic/beats/metricbeat/module/redis/key/data.go:28
github.com/elastic/beats/metricbeat/module/redis/key.(*MetricSet).Fetch
    /go/src/github.com/elastic/beats/metricbeat/module/redis/key/key.go:98
github.com/elastic/beats/metricbeat/mb/module.(*metricSetWrapper).fetch
    /go/src/github.com/elastic/beats/metricbeat/mb/module/wrapper.go:238
github.com/elastic/beats/metricbeat/mb/module.(*metricSetWrapper).startPeriodicFetching
    /go/src/github.com/elastic/beats/metricbeat/mb/module/wrapper.go:219
github.com/elastic/beats/metricbeat/mb/module.(*metricSetWrapper).run
    /go/src/github.com/elastic/beats/metricbeat/mb/module/wrapper.go:196
github.com/elastic/beats/metricbeat/mb/module.(*Wrapper).Start.func1
    /go/src/github.com/elastic/beats/metricbeat/mb/module/wrapper.go:137"}
```

The Redis module config we're using:
```
- module: redis
  metricsets: ["key"]
  key.patterns:
    - pattern: '*'
  period: 5s

  processors:
    - drop_event:
        when:
          contains:
            redis.key.name: "_kombu"
    - drop_event:
        when:
          contains:
            redis.key.name: "unack"

  # Redis hosts
  hosts: ["xxxxx:6379"]
```

**Comment by jsoriano** [comment]
Hey @jwarren116,

Thanks for reporting this. Actually this fix is merged in the 6.8 branch, but still unreleased, in principle it will get released in 6.8.4. 6.8.3 doesn't contain the fix. Sorry for the confusion caused, I though when preparing the fix that it will be in time for 6.8.3.

**Comment by jwarren116** [comment]
Ahh, that makes sense. Thank you!
